ui/dialog.js

```diff
@@ -412,20 +413,21 @@ return $.widget( "ui.dialog", {
 				},
 				text: false
 			})
-			.addClass( "ui-dialog-titlebar-close" )
 			.appendTo( this.uiDialogTitlebar );
-		this._on( this.uiDialogTitlebarClose, {
-			click: function( event ) {
-				event.preventDefault();
-				this.close( event );
-			}
-		});
 
-		uiDialogTitle = $( "<span>" )
+		this._addClass( this.uiDialogTitlebarClose, "ui-dialog-titlebar-close" )
```

Comment:
Actually, just don't chain. We're not chaining for classes and events in other places.